feature/policy_forwarding/encapsulation/otg_tests/decap_gre_ipv4/decap_gre_ipv4_test.go

```diff
@@ -0,0 +1,836 @@
+		fptest.AssignToNetworkInstance(t, dut, p2.Name(), deviations.DefaultNetworkInstance(dut), 0)
+	}
+
+	if deviations.ExplicitPortSpeed(dut) {
+		fptest.SetPortSpeed(t, p1)
+		fptest.SetPortSpeed(t, p2)
+	}
+}
+
+// Configures the given DUT interface.
+func configInterfaceDUT(p *ondatra.Port, a *attrs.Attributes, dut *ondatra.DUTDevice) *oc.Interface {
+	i := a.NewOCInterface(p.Name(), dut)
+	s4 := i.GetOrCreateSubinterface(0).GetOrCreateIpv4()
+	if deviations.InterfaceEnabled(dut) && !deviations.IPv4MissingEnabled(dut) {
+		s4.Enabled = ygot.Bool(true)
+	}
+	i.GetOrCreateSubinterface(0).GetOrCreateIpv6()
+
+	return i
+}
+
+// Congigure Static Routes on DUT
+func configStaticRoute(t *testing.T, dut *ondatra.DUTDevice, prefix string, nexthop string) {
+	s := &oc.Root{}
+	ni := s.GetOrCreateNetworkInstance(deviations.DefaultNetworkInstance(dut))
+	static := ni.GetOrCreateProtocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_STATIC, deviations.StaticProtocolName(dut))
+	nh := static.GetOrCreateStatic(prefix).GetOrCreateNextHop("0")
+	if nexthop == "Null0" {
+		nh.SetNextHop(oc.UnionString("DROP"))
+	} else {
+		nh.NextHop = oc.UnionString(nexthop)
+	}
+
+	gnmi.Update(t, dut, gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).Config(), ni)
+}
+
+// Configure static MPLS label binding (LBL1) towards ATE Port 2
+func configureStaticMPLSCLI(t *testing.T, dut *ondatra.DUTDevice) {
+	if deviations.StaticMplsLspUnsupported(dut) {
+		switch dut.Vendor() {
+		case ondatra.ARISTA:
+			gnmiClient := dut.RawAPIs().GNMI(t)
+
+			jsonConfig := fmt.Sprintf(`
+				mpls ip
+				mpls static top-label %v %s %s pop payload-type ipv4
+				`, LBL1, dut.Port(t, "port2").Name(), otgPort2.IPv4,
+			)
+
+			gpbSetRequest := buildCliConfigRequest(jsonConfig)
+
+			if _, err := gnmiClient.Set(context.Background(), gpbSetRequest); err != nil {
+				t.Fatalf("gnmiClient.Set() with unexpected error: %v", err)
+			}
+		default:
+			t.Errorf("Deviation StaticMplsLspUnsupported is not handled for the dut: %v", dut.Vendor())
+		}
+	} else {
+		d := &oc.Root{}
+		fptest.ConfigureDefaultNetworkInstance(t, dut)
+		mplsCfg := d.GetOrCreateNetworkInstance(deviations.DefaultNetworkInstance(dut)).GetOrCreateMpls()
+		staticMplsCfg := mplsCfg.GetOrCreateLsps().GetOrCreateStaticLsp("lsp1")
+		staticMplsCfg.GetOrCreateEgress().SetIncomingLabel(oc.UnionUint32(LBL1))
+		staticMplsCfg.GetOrCreateEgress().SetNextHop(otgPort2.IPv4)
+		staticMplsCfg.GetOrCreateEgress().SetPushLabel(oc.Egress_PushLabel_IMPLICIT_NULL)
+
+		gnmi.Update(t, dut, gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).Mpls().Config(), mplsCfg)
+	}
+}
+
+// GRE traffic with destination DECAP-DST using destination-address-prefix-set and decapsulate.
+// Adding deviation, currently gre decapsulation is not supported, using CLI.
+func configureGRETunnel(t *testing.T, dut *ondatra.DUTDevice) {
+	if deviations.GreDecapsulationUnsupported(dut) {
```

Comment:
indicate the OC for Gre decap is not supported, but the feature itself is supported using some alternate OC or CLI.
```suggestion
	if deviations.GreDecapsulationOCUnsupported(dut) {
```